app/controllers/admin/loaned_books_controller.rb

```diff
@@ -73,4 +86,27 @@ def check_valid_user
     flash.now[:danger] = t "error.user_invalid"
     render :new
   end
+
+  def check_status_params
+    status_value = LoanedBook.statuses.values
+    return if status_value.include?(params[:status].to_i)
+
+    render json: {error: I18n.t("error.status_not_exist")}
+  end
+
+  def update_status_loaned_detail params
+    @loaned_book.loaned_details.each {|loan| loan.update!(status: params)}
```

Comment:
```suggestion
    @loaned_book.loaned_details.each {|loan_detail| loan_detail.update!(status: params) unless loan_detail.cancel?}
```
Cái này thực hiện update status cho detail với dk detail ko phải là cancel chứ cu hì